rfcs/text/0009_screenshot_mode_service.md

```diff
@@ -0,0 +1,262 @@
+
+When Kibana loads initially, there is a Newsfeed component in the UI that
+checks internally cached records to see if it must fetch the Elastic News
+Service for newer items. When the Screenshot Mode Service is implemented, the
+Newsfeed component has a source of information to check on whether or not it
+should load in the Kibana UI. If it can avoid loading, it avoids an unnecessary
+HTTP round trip, which weigh heavily on performance.
+
+# Motivation
+
+Kibana Reporting is a commercial feature in Elastic and is highly capable of
+loading the application pages and converting them into a screenshot for PNG or
+PDF export. However, the way it works has downsides with performance,
+maintainability, and expanding it as a tool that can power higher-level
+features. The solution to those downsides is to have Kibana pages themselves
+become more capable at presenting themselves for screenshot capture reports.
+With the Screenshot Mode Service available, Reporting could drop the
+task that it currently has which hurt performance: wasted rendering that is
+replaced with custom styles that make the page "reportable."
+
+The technical advantage of having such as service also leads to making Kibana
+application pages "printable", in the sense that sending the page to a printer
+for a hard copy results in something more meaningful and specialized for paper
+than today's Kibana can guarantee. This isn't a big concern for Kibana since
+there isn't the expectation to improve printing Kibana, but that technical
+direction is appropriate for improving PDF report generation.
+
+# Detailed design
+
+The Screenshot Mode Service is a callable API available in the context of a
+request on the server side, and as a pubic pluginsSetup Javascript API.
+
+The data provided by the Screenshot Mode are signals about:
+ - whether or not the context of the page load is for a screenshot capture
+ - layout and dimensions that the result image is expected to be
+ - time zone for which to format the page data
+
+To obtain the screenshot mode context data and provide it as a service to
+plugins, this RFC recommends using a URL query string variable with raw data,
+and provide an interface to it in the form of an observable. The Screenshot
+Mode Service plugin will read the data in its `setup` method and provide the
+observable to the data interface in its plugin contract. The interface is
+described below as the `ScreenshotModeServiceApi`.
+
+The new query string parameter could appear in the URL of any Kibana
+application. Adding new URL query string parameters is a non-breaking change,
+and applications would not be impacted by having this additional data in the URL.
+
+Since the service data needs to be an object, the best way to encode it as data
+in a URL is to use Rison encoding. An example of what that would look like in a
+Kibana URL (with linebreaks added):
+
+```
+http://localhost:5601/app/kibana
+?screenshots=(enabled:!t,layout:(height:500,width:400),timezone:PST)
```

Comment:
who is going to parse this query in OSS mode?

Reply:
The Screenshot Mode Service will be a plugin in OSS Kibana.